app/routes.py

```diff
@@ -1 +1,280 @@
-from flask import Blueprint
\ No newline at end of file
+import os
+from datetime import datetime
+
+import requests
+from flask import Blueprint, abort, jsonify, make_response, request
+
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+
+task_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+##-----------------------------------------------------------------##
+##--------------------------Helper Functions-----------------------##
+##-----------------------------------------------------------------##
+
+def validate_item(cls,id):
+    try:
+        id = int(id)
+    except:
+        abort(make_response({"message": f"{cls.__name__} {id} is invalid"}, 400))
+
+    item = cls.query.get(id)
+    if not item:
+        abort(make_response({"message": f"{cls.__name__} {id} not found"}, 404))
+
+    return item
+
+def validate_new_item(cls,request_body):
+    try:  
+        if cls == Task:
+            new_item = cls(title=request_body["title"],
+                           description=request_body["description"]
+                        )
+        else:
+             new_item = cls(title=request_body["title"])
+    except:
+        abort(make_response({"details": "Invalid data"},400))
+
+    return new_item
+
+def post_to_slack(task):
+    URL = "https://slack.com/api/chat.postMessage"
+    token = os.environ.get('SLACK_TOKEN')
+    Headers = {"Authorization" : f'Bearer {token}'}
+    params = {'channel':'task-notifications',
+              'text': f'Someone just completed the task {task.title}'
+            }
+
+    requests.post(URL,headers=Headers,params=params)
+
+def task_to_goal(task_id,goal):
+    task = validate_item(Task,task_id)
+    task.goal_id=goal.goal_id
+
+    return task_id
+
+def to_dict(self,list_tasks=False):
```

Comment:
This is an interesting approach to creating a `to_dict` function that works for both of these models. I'd point out that this might not be the most extensible approach, since for any additional models you want to handle in this function will increase its size, complexity, and readability considerably.